tests/test_serialization.py

```diff
@@ -0,0 +1,202 @@
+# Copyright 2025
+#
+#    Licensed under the Apache License, Version 2.0 (the "License");
+#    you may not use this file except in compliance with the License.
+#    You may obtain a copy of the License at
+#
+#        http://www.apache.org/licenses/LICENSE-2.0
+#
+#    Unless required by applicable law or agreed to in writing, software
+#    distributed under the License is distributed on an "AS IS" BASIS,
+#    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#    See the License for the specific language governing permissions and
+#    limitations under the License.
+
+"""
+This file is designed for backwards compatibility! So it tries to use
+the most stable APIs.
+Specifically, it needs to work with dwave-optimization 0.5.2, the last
+version that only supported serialization version 0.1
+"""
+
+from __future__ import annotations
+
+import os.path
+import tempfile
+import typing
+import unittest
+
+import numpy as np
+
+import dwave.optimization
+from dwave.optimization import Model
+
+
+class TestSerialization(unittest.TestCase):
+    """For each model to be tested we expect a ``make_<model>()`` method."""
+    def assertModelEqual(self, lhs: Model, rhs: Model):
+        """Assert two models are the same."""
+        with lhs.lock(), rhs.lock():
+            # The models must have the same number of symbols
+            self.assertEqual(lhs.num_symbols(), rhs.num_symbols())
+
+            # The models must have the same number of states
+            self.assertEqual(lhs.states.size(), rhs.states.size())
+
+            # All nodes in the model need to match, and to have the same states
+            for ls, rs in zip(lhs.iter_symbols(), rhs.iter_symbols()):
+                self.assertTrue(ls.maybe_equals(rs))
```

Comment:
Why not the full equality check? Or the full check if the result here is `MAYBE`?